src/lib.rs

```diff
@@ -8,7 +8,156 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! A typesafe bitmask flag generator.
+//! A typesafe bitmask flag generator useful for sets of C-style bitmask flags.
+//! It can be used for creating typesafe wrappers around C APIs.
+//!
+//! The `bitflags!` macro generates a `struct` that manages a set of flags. The 
+//! flags should only be defined for integer types, otherwise unexpected type 
+//! errors may occur at compile time.
+//!
+//! # Example
+//!
+//! ```{.rust}
```

Comment:
don't include the `{.rust}` bit; this is an older style, today it would be

    ```rust

but, we don't include the `rust` part in API docs, so just triple graves :smile: